flight0handler.go

```diff
@@ -0,0 +1,89 @@
+package dtls
+
+import (
+	"context"
+	"crypto/rand"
+)
+
+func flight0Parse(ctx context.Context, c flightConn, state *State, cache *handshakeCache, cfg *handshakeConfig) (flightVal, *alert, error) {
+	// HelloVerifyRequest can be skipped by the server,
+	// so allow ServerHello during flight1 also
+	seq, msgs, ok := cache.fullPullMap(state.handshakeRecvSequence,
```

Comment:
I would put `!ok` first.

I know `fullPullMap` always gives a value valid for `seq`, but I needed to confirm. When I see `!ok` I assume all return values are no good.

Reply:
What do you think of moving the `flight requirements` out of the Parse functions?

I am sure it is fine, but seems fragile to do the checking/asserting in every function. Would be nice if each Parse function was called with everything it needed.

My vote is that we fix this in a follow up PR, but merge what you have now :)

Reply:
I would just be really nice if these functions weren't re-entrant.

Reply:
> What do you think of moving the flight requirements out of the Parse functions?

One exception making it complexed is `flight4Parse`.
Flight4 contains `Finished` message, that it is not available to be decrypted until parsing `ClientKeyExchange`.